pkg/mutation/schema/schema.go

```diff
@@ -0,0 +1,398 @@
+package schema
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path/parser"
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/types"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// TODO we are assuming that remove will never get corrupted.
+// this should be the case, but is that wise?
+
+// Binding represent the specific GVKs that a
+// mutation's implicit schema applies to
+type Binding struct {
+	Groups   []string
+	Kinds    []string
+	Versions []string
+}
+
+// MutatorWithSchema is a mutator exposing the implied
+// schema of the target object.
+type MutatorWithSchema interface {
+	types.Mutator
+	SchemaBindings() []Binding
+}
+
+var (
+	log = logf.Log.WithName("mutation_schema")
+)
+
+type bindingIterator struct {
+	sorted bool
```

Comment:
Is the sorted flavour ever used?
If not, can we simplify this?

Reply:
only for unit tests. Maybe we should always sort? I'm saving `N` vs `Nlog(N)` and `N` is probably not that large to start with.